javaparser-core/src/main/java/com/github/javaparser/ast/type/ArrayType.java

```diff
@@ -0,0 +1,26 @@
+package com.github.javaparser.ast.type;
+
+import com.github.javaparser.Range;
+import com.github.javaparser.ast.expr.AnnotationExpr;
+import com.github.javaparser.ast.visitor.GenericVisitor;
+import com.github.javaparser.ast.visitor.VoidVisitor;
+
+import java.util.List;
+
+/**
+ * Every type followed by [] gets wrapped in an ArrayType for each [].
+ */
+public class ArrayType extends ReferenceType {
+    public ArrayType(Range range, Type type, List<AnnotationExpr> annotations) {
```

Comment:
Added a commit that does step 1: make ReferenceType "just a base class" so the cast works. Saves a huge amount of nodes in the AST!